sql/src/main/java/io/crate/executor/transport/ExecutionNodesTask.java

```diff
@@ -82,12 +83,13 @@
     private TransportCloseContextNodeAction transportCloseContextNodeAction;
     private final StreamerVisitor streamerVisitor;
     private final CircuitBreaker circuitBreaker;
-    private MergeNode mergeNode;
+    private List<MergeNode> mergeNodes;
+    private boolean rowCountResult = false;
 
     /**
-     * @param mergeNode the mergeNode for the final merge operation on the handler.
+     * @param mergeNodes list of mergeNodes for the final merge operation on the handler.
```

Comment:
maybe extend the comment here to clarify that multiple final merge operations are only the case if there is a bulk operation happening